### ISSUE TITLE
Docs: Confirm block comment approach for style guide

### ISSUE THREAD

**Issue by aubergine10** [eslint/eslint#6518]
> Note: I'm happy to manually update literally every rule doc over the coming weeks to make the block comments consistent. In this discussion I would suggest prioritising "What is best for the reader" rather than "What is least work for @aubergine10".
## Spacing

It's become clear that we should describe how eslint block comments should be spaced to ensure consistency throughout documentation.

My personal preference is nitrogen-packed as follows:

```
/* eslint rule-name: [ "off", { "prop": val } ] */ 
```

I feel this greatly enhances legibility, compare to the suction-packed version:

```
/*eslint rule-name: ["off", {"prop": val}]*/ 
```

While it takes up less space, it's more difficult to read IMO. In particular, comments don't get syntax highlighting to differentiate brackets, quotes, strings, numbers, etc.
## Quoting

Should the rule-name be "double-qouted"?

My personal preference is that it shouldn't (looks cleaner without quotes), but not doing so breaks consistency compared to the `.eslintrc` and `package.json`.

There is also consistency between the rule name and the options, eg. in the example above `"prop"` is quoted. Arguably, if we're not quoting the rule name, the prop keys shouldn't be quoted either.

There's also the argument that because prop names include non-identifier characters, they should be quoted.

See [quote-props rule](http://eslint.org/docs/rules/quote-props) for potential approaches we could consider.

Discuss :)

**Comment by platinumazure** [comment]
My opinion is we should try to follow what the linting rules are for JavaScript (in our repository), so that anyone writing code and documentation would write the same style of "JSON" in the inline comments.

Until recently, I thought that meant no spaces in array brackets, but spaces in object curly braces. Lately, though, I've seen suction-packed objects, though, so either we turned off the rule or my memory is just wrong.

**Comment by aubergine10** [comment]
I agree, to an extent. I don't think we can treat code linting the same as comment linting - in code you've got syntax highlighting which makes a huge difference in terms of legibility. In block comments, all the text is one color - legibility (or lack of) is largely down to spacing.

**Comment by aubergine10** [comment]
Another thing to consider is that (and I've tested to confirm) the eslint comments can be multiline, just like their `.eslintrc` and `package.json` counterparts:

```
/* eslint "quote-props": [ "off", "always", {
  "keywords": false,
  "numbers": false,
  "unnecessary": true
} ] */
```

**Comment by nzakas** [comment]
I don't find either format easier or harder to read than the other, so I'll happily go along with whatever the consensus is. I have a slight preference for the second, more compressed version, `/*eslint rule-name: "error"*/`, as I think the first has a bit too much space, but that's more of a preference than anything else.

I do think we should omit quotes around rule names and option names.

**Comment by aubergine10** [comment]
Examples with no quotes - much cleaner:

```
/* eslint rule-name: [ off, { prop: val } ] */ 

/*eslint rule-name: [off, {prop: val}]*/ 
```

**Comment by nzakas** [comment]
I think that "off" and "val" still need to be in quotes (?). Even if not required, I'd like them to have quotes. I just meant I think the rulename shouldn't have quotes (that's generally been what we've done all along - rulenames have no quotes, other stuff does).

**Comment by aubergine10** [comment]
`off` - agreed, it's a string literal. `val` - only if it's a string (it could be boolean, array, etc).
```
/* eslint rule-name: [ "off", { prop: val } ] */ 

/*eslint rule-name: ["off", {prop: val}]*/ 
```

**Comment by nzakas** [comment]
Yeah, `val` should be quoted in your example because it's a string. Numeric or boolean literals would not be quoted.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
